internal/declcfg/declcfg_to_model.go

```diff
@@ -0,0 +1,113 @@
+package declcfg
+
+import (
+	"fmt"
+
+	"github.com/operator-framework/operator-registry/internal/model"
+	"github.com/operator-framework/operator-registry/internal/property"
+)
+
+func ConvertToModel(cfg DeclarativeConfig) (model.Model, error) {
+	mpkgs := model.Model{}
+	defaultChannels := map[string]string{}
+	for _, p := range cfg.Packages {
+		mpkg := &model.Package{
+			Name:        p.Name,
+			Description: p.Description,
+			Channels:    map[string]*model.Channel{},
+		}
+		if p.Icon != nil {
+			mpkg.Icon = &model.Icon{
+				Data:      p.Icon.Data,
+				MediaType: p.Icon.MediaType,
+			}
+		}
+		defaultChannels[p.Name] = p.DefaultChannel
+		mpkgs[p.Name] = mpkg
+	}
+
+	for _, b := range cfg.Bundles {
+		defaultChannelName := defaultChannels[b.Package]
+		if b.Package == "" {
+			return nil, fmt.Errorf("package name must be set for bundle %q", b.Name)
+		}
+		mpkg, ok := mpkgs[b.Package]
+		if !ok {
+			return nil, fmt.Errorf("unknown package %q for bundle %q", b.Package, b.Name)
+		}
+
+		props, err := parseProperties(b.Properties)
+		if err != nil {
+			return nil, fmt.Errorf("parse properties for bundle %q: %v", b.Name, err)
+		}
+
+		if b.Package != props.Packages[0].PackageName {
+			return nil, fmt.Errorf("package %q does not match %q property %q", b.Package, property.TypePackage, props.Packages[0].PackageName)
+		}
+
+		for _, bundleChannel := range props.Channels {
```

Comment:
This doesn't account for the case where channel property is missing and this bundle is orphaned. Will fix later.